OpenRA.Mods.Common/Traits/Render/WithMakeAnimation.cs

```diff
@@ -20,36 +21,79 @@ public class WithMakeAnimationInfo : ITraitInfo, Requires<WithSpriteBodyInfo>
 		[Desc("Sequence name to use")]
 		[SequenceReference] public readonly string Sequence = "make";
 
+		[GrantedConditionReference]
+		[Desc("The condition to grant to self while the make animation is playing.")]
+		public readonly string Condition = null;
+
 		public object Create(ActorInitializer init) { return new WithMakeAnimation(init, this); }
 	}
 
-	public class WithMakeAnimation
+	public class WithMakeAnimation : INotifyCreated
 	{
 		readonly WithMakeAnimationInfo info;
 		readonly WithSpriteBody wsb;
+		int token = ConditionManager.InvalidConditionToken;
 
 		public WithMakeAnimation(ActorInitializer init, WithMakeAnimationInfo info)
 		{
 			this.info = info;
 			var self = init.Self;
 			wsb = self.Trait<WithSpriteBody>();
+		}
 
+		void INotifyCreated.Created(Actor self)
+		{
 			var building = self.TraitOrDefault<Building>();
 			if (building != null && !building.SkipMakeAnimation)
+				Forward(self, () => building.NotifyBuildingComplete(self));
+		}
+
+		void Forward(Actor self, Action onComplete)
+		{
+			var conditionManager = self.TraitOrDefault<ConditionManager>();
+			if (conditionManager != null && !string.IsNullOrEmpty(info.Condition) && token == ConditionManager.InvalidConditionToken)
+				token = conditionManager.GrantCondition(self, info.Condition);
```

Comment:
I'm now caching the condition manager, but i'd prefer not to merge the info.Condition check into the manager query because that breaks the standard pattern that i'd like to keep (it won't work if a trait grants multiple conditions).